### ISSUE TITLE
Coding guidelines, naming conventions and style.

### ISSUE THREAD

**Issue by Mihaylov93** [SimulIDE/SimulIDE#7]
To ensure code quality and consistency, adopting from the early stages some forms of conventions would help on the long run.
For coding guidelines, there is not much to discuss here, https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines 

About the naming conventions, there is some historical conventions but nothing is standard, here I present what I usually use based on work/personal projects:

### prefixes
- **_** variable For private class variables.
- **m**Variable For member variables.
- **i/o/io**Variable to indicate if  method arguments are input, output or input output.
- **k** For globals, e.g. defines.

For the style we should use a styling tool like astyle or clang-format, from clang-format we could choose one of the predefined styles (google, webkit...) or create a custom.

Here you can find the formatting style I personally use, based on webkit https://gist.github.com/Mihaylov93/fd302d5ce154b82418f42b6f98f1500d

**Comment by Foadsf** [comment]
@Mihaylov93 thanks a lot. assigned it to yourself. the more we go forward, I will give you guys write access and other moderation privileges to the repo.

**Comment by pratikpc** [comment]
@Mihaylov93 we could also add Clang-Tidy or Clazy

**Comment by Mihaylov93** [comment]
Yes, I always use it locally. It is a bit harder to use as an action than just clang-format, but using it locally before pushing is always welcome.
